src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -673,6 +673,11 @@ private async Task<ConnectBlocksResult> FullyValidateLockedAsync(ChainedHeader n
 
             // Add peers that needed to be banned as a result of a failure to connect blocks.
             // Otherwise they get lost as we are returning a different ConnnectBlocksResult.
+            // We also need to set the ban reason and ban time otherwise it is not known why
+            // connecting the new chain failed and hence why the peer is being disconnected in 
+            // peer banning.
+            reconnectionResult.BanReason = connectBlockResult.BanReason;
+            reconnectionResult.BanDurationSeconds = connectBlockResult.BanDurationSeconds;
```

Comment:
This looks correct to me but could probably use a second opinion.

I note that the ban duration would fall back to the same default value that it would otherwise get from the validation context, so that should be a harmless addition that merely makes the intended result more explicit.